File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/ButtonBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;
import org.sandboxpowered.api.state.property.EnumProperty;

public class ButtonBlock extends BaseBlock {
    private final EnumProperty<AttachFace> FACE = EnumProperty.of("face", AttachFace.class);

    public ButtonBlock(Settings settings) {
        super(settings);
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);
        builder.add(Properties.HORIZONTAL_FACING, Properties.POWERED, FACE);
    }

    public enum AttachFace {
        FLOOR,
        WALL,
        CEILING
    }

}

File: common/src/main/kotlin/org/sandboxpowered/silica/block/ButtonBlock.kt
package org.sandboxpowered.silica.block

import org.sandboxpowered.silica.block.SilicaBlockProperties.BUTTON_FACE
import org.sandboxpowered.silica.block.SilicaBlockProperties.HORIZONTAL_FACING
import org.sandboxpowered.silica.block.SilicaBlockProperties.POWERED
import org.sandboxpowered.silica.state.StateProvider
import org.sandboxpowered.silica.state.block.BlockState
import org.sandboxpowered.silica.util.Identifier

class ButtonBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(POWERED, HORIZONTAL_FACING, BUTTON_FACE)
    }
}